lib/offsite_payments/integrations/molpay.rb

```diff
@@ -0,0 +1,246 @@
+    module Molpay
+
+      mattr_accessor :acknowledge_url 
+
+      self.acknowledge_url = 'https://www.onlinepayment.com.my/MOLPay/API/chkstat/returnipn.php' 
+
+      def self.notification(post)
+        Notification.new(post)
+      end
+
+      def self.return(query_string, options={})
+        Return.new(query_string, options)
+      end
+
+      # Example.
+      #
+      #  (Optional Parameter) = channel //will generate URL to go directly to specific channel, e.g maybank2u, cimb 
+      #  Please refer MOLPay API spec for the channel routing
+      #
+      #  payment_service_for('ORDER_ID', 'MOLPAY_MERCHANT_ID', :service => :molpay,  :amount => 1.01, :currency => 'MYR', :credential2 => 'MOLPAY_VERIFICATION_KEY', :channel => 'maybank2u.php') do |service|
+      #
+      #    service.customer :name              => 'Your Name',
+      #                     :email             => 'name@molpay.com',
+      #                     :phone             => '60355218438'
+      #
+      #    service.description "Payment for Item 001"
+      #
+      #    service.country "MY"
+      #
+      #    service.language "en"
+      #
+      #    service.return_url 'http://yourstore.com/return'
+      #
+      # end
+      #
+      class Helper < OffsitePayments::Helper
+        
+        #Currencies supported
+        #MYR (Malaysian Ringgit - Malaysia Payment Gateway (Credit Card & local debit payment), Union Pay, Alipay)
+        #USD (US Dollar)
+        #SGD (Singapore Dollar)
+        #PHP (Philippines Peso)
+        #VND (Vietnamese Dong)
+        #IDR (Indonesian Rupiah)
+        #AUD (Australian Dollar)
+        SUPPORTED_CURRENCIES = [ 'MYR', 'USD', 'SGD', 'PHP', 'VND', 'IDR', 'AUD']
+
+        #Languages supported
+        #en English (default)
+        #cn Simplified Chinese
+        SUPPORTED_LANGUAGES = ['en', 'cn']
+
+        SERVICE_URL = 'https://www.onlinepayment.com.my/MOLPay/pay/'.freeze
+
+        mapping :account, 'merchantid'
+        mapping :amount, 'amount'
+        mapping :order, 'orderid'
+        mapping :customer, :name  => 'bill_name',
+                           :email => 'bill_email',
+                           :phone => 'bill_mobile'
+
+        mapping :description, 'bill_desc'
+        mapping :language, 'langcode'
+        mapping :country, 'country'
+        mapping :currency, 'cur'
+        mapping :return_url, 'returnurl'
+        mapping :signature, 'vcode'
+
+
+        attr_reader :amount_in_cents, :verify_key, :channel
+
+        def credential_based_url
+          service_url = SERVICE_URL + @fields[mappings[:account]] + "/"
+          service_url = service_url + @channel unless @channel.blank?
+          service_url
+        end
+
+        def initialize(order, account, options = {})
+          @verify_key = options[:credential2] if options[:credential2]
+          @amount_in_cents = options[:amount]
+          @channel = options[:channel] if options[:channel]
+          options.delete(:channel)
```

Comment:
`#delete` returns the deleted value, or nil if it's not found, so this line and the one above it can be combined: @channel = options.delete(:channel)`